inc/agent.class.php

```diff
@@ -1091,12 +1096,17 @@ protected function enrollByInvitationToken($input) {
       // Check the serial does not exists in the entity
       $computer = new Computer();
       $entityId = $invitation->getField('entities_id');
-      $rows = $computer->find("`entities_id`='$entityId' AND `serial`='$serial'");
-      if (count($rows) > 0) {
-         $event = __('The serial already exists', 'flyvemdm');
-         $this->filterMessages($event);
-         $this->logInvitationEvent($invitation, $event);
-         return false;
+      $computers = $computer->find("`entities_id`='$entityId' AND `serial`='$serial'");
+      if (count($computers) > 0) {
+         // A computer with the same serial exist, let's find if an agent exist for that device.
+         $flyvemdmAgent = new PluginFlyvemdmAgent();
+         $agents = $flyvemdmAgent->find("`entities_id`='$entityId' AND `computers_id`='{$computer->getID()}'");
```

Comment:
{$computer->getID()} <=== the syntax {$obj->method()} is too messy. I mrefer using a var to store the value of the method then use it here.

Reply:
ok